Sources/Classes/RSCloudModeManager.m

```diff
@@ -38,51 +38,55 @@ - (void) startCloudModeProcessor {
+                } else if (response.state == INVALID_URL) {
+                    [RSLogger logError:@"RSCloudModeManager: CloudModeProcessor: Invalid Data Plane URL. Aborting the Cloud Mode Processor"];
+                    [RSMetricsReporter report:SDKMETRICS_CM_ATTEMPT_ABORT forMetricType:COUNT withProperties:@{SDKMETRICS_TYPE: SDKMETRICS_DATA_PLANE_URL_INVALID} andValue:1];
+                    break;
+                } else if (response.state == NETWORK_ERROR) {
+                    int delay = [self->backOff nextDelay];
+                    [RSLogger logDebug:[[NSString alloc] initWithFormat:@"RSCloudModeManager: CloudModeProcessor: Retrying in: %@", [RSUtils secondsToString:delay]]];
+                    [RSMetricsReporter report:SDKMETRICS_CM_ATTEMPT_RETRY forMetricType:COUNT withProperties:nil andValue:1];
+                    sleep(delay);
+                } else { // To handle the status code RESOURCE_NOT_FOUND(404) & BAD_REQUEST(400)
+                    [RSLogger logDebug:[[NSString alloc] initWithFormat:@"RSCloudModeManager: CloudModeProcessor: Retrying in: 1s"]];
+                    sleep(1);
+                }
             }
-            [strongSelf->lock unlock];
-            [RSLogger logDebug:[[NSString alloc] initWithFormat:@"RSCloudModeManager: CloudModeProcessor: cloudModeSleepCount: %d", sleepCount]];
-            sleepCount += 1;
-            
-            if(response == nil) {
-                sleep(1);
-            } else if (response.state == WRONG_WRITE_KEY) {
-                [RSLogger logError:@"RSCloudModeManager: CloudModeProcessor: Wrong WriteKey. Aborting the Cloud Mode Processor"];
-                break;
-            } else if (response.state == INVALID_URL) {
-                [RSLogger logError:@"RSCloudModeManager: CloudModeProcessor: Invalid Data Plane URL. Aborting the Cloud Mode Processor"];
-                [RSMetricsReporter report:SDKMETRICS_CM_ATTEMPT_ABORT forMetricType:COUNT withProperties:@{SDKMETRICS_TYPE: SDKMETRICS_DATA_PLANE_URL_INVALID} andValue:1];
-                break;
-            } else if (response.state == NETWORK_ERROR) {
-                int delay = [self->backOff nextDelay];
-                [RSLogger logDebug:[[NSString alloc] initWithFormat:@"RSCloudModeManager: CloudModeProcessor: Retrying in: %@", [RSUtils secondsToString:delay]]];
-                [RSMetricsReporter report:SDKMETRICS_CM_ATTEMPT_RETRY forMetricType:COUNT withProperties:nil andValue:1];
-                sleep(delay);
-            } else { // To handle the status code RESOURCE_NOT_FOUND(404) & BAD_REQUEST(400)
-                [RSLogger logDebug:[[NSString alloc] initWithFormat:@"RSCloudModeManager: CloudModeProcessor: Retrying in: 1s"]];
-                sleep(1);
+            @catch (NSException *exception) {
+                [RSLogger logError:[NSString stringWithFormat:@"RSCloudModeManager: CloudModeProcessor: Failed to flush current batch, reason: %@", exception.reason]];
             }
         }
```

Comment:
Please test the changes thoroughly:
```suggestion
            @catch (NSException *exception) {
                [strongSelf->lock unlock];
                [RSLogger logError:[NSString stringWithFormat:@"RSCloudModeManager: CloudModeProcessor: Failed to flush current batch, reason: %@", exception.reason]];
                sleep(1);
            }
        }
```

Currently, if any exception occurs, the lock is unreleased and CloudModeManager stops working and also it is better to have a one-second delay so that the CPU waits for some time before attempting again.